src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckVulnerabilityMessageContentCollector.java

```diff
@@ -71,35 +82,72 @@ protected void addCategoryItems(final List<CategoryItem> categoryItems, final Js
         final LinkableItem componentVersionItem = componentVersionItems.stream().findFirst().orElse(null);
 
         final Class<VulnerabilitySourceQualifiedId> targetClass = VulnerabilitySourceQualifiedId.class;
+
         try {
             final List<VulnerabilitySourceQualifiedId> newVulnerabilityList = getFieldValueObjectsByLabel(jsonFieldAccessor, vulnerabilityFields, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_NEW, targetClass);
-            addVulnerabilityItems(categoryItems, notificationContent.getId(), ItemOperation.ADD, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_NEW, componentItem, componentVersionItem, newVulnerabilityList);
-
             final List<VulnerabilitySourceQualifiedId> updatedVulnerabilityList = getFieldValueObjectsByLabel(jsonFieldAccessor, vulnerabilityFields, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_UPDATED, targetClass);
-            addVulnerabilityItems(categoryItems, notificationContent.getId(), ItemOperation.UPDATE, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_NEW, componentItem, componentVersionItem, updatedVulnerabilityList);
-
             final List<VulnerabilitySourceQualifiedId> deletedVulnerabilityList = getFieldValueObjectsByLabel(jsonFieldAccessor, vulnerabilityFields, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_DELETED, targetClass);
-            addVulnerabilityItems(categoryItems, notificationContent.getId(), ItemOperation.DELETE, BlackDuckProviderContentTypes.LABEL_VULNERABILITY_NEW, componentItem, componentVersionItem, deletedVulnerabilityList);
-        } catch (final AlertException e) {
-            logger.error("Mishandled the expected type of a notification field", e);
+            final Optional<BlackduckRestConnection> restConnectionOptional = blackDuckProperties.createRestConnectionAndLogErrors(logger);
+            // TODO: this seems overly complicated to get a service connection to BlackDuck.  Need to implement a component that manages the connections and ask that class for a service.
```

Comment:
The problem with creating a Component to create the services is, how/when do we close the BlackDuckRestConnection? At least this way we know we aren't leaving open connections